test/integration/uds_integration_test.cc

```diff
@@ -78,6 +78,31 @@ HttpIntegrationTest::ConnectionCreationFunction UdsListenerIntegrationTest::crea
   };
 }
 
+TEST_P(UdsListenerIntegrationTest, TestPeerCredentials) {
+  fake_upstreams_count_ = 1;
+  initialize();
+  auto client_connection = createConnectionFn()();
+  codec_client_ = makeHttpConnection(std::move(client_connection));
+  Http::TestHeaderMapImpl request_headers{
+      {":method", "POST"},    {":path", "/test/long/url"}, {":scheme", "http"},
+      {":authority", "host"}, {"x-lyft-user-id", "123"},   {"x-forwarded-for", "10.0.0.1"}};
+  auto response = codec_client_->makeHeaderOnlyRequest(request_headers);
+  waitForNextUpstreamRequest(0);
+
+  auto credentials = codec_client_->connection()->peerCredentials();
+#ifndef SO_PEERCRED
+  EXPECT_EQ(absl::nullopt, credentials);
```

Comment:
nit: prefer to put the tested thing first, eg, `EXPECT_EQ(credentials, absl::nullopt)`.

I was hoping there was an `absl::nullopt` variant of `EXPECT_THAT(credentials, IsNull())` existed since it reads really clearly but I can't find it.